.github/workflows/pull_request.yml

```diff
@@ -5,6 +5,16 @@ on:
       - main
 
 jobs:
+  check_python_linting:
+    name: Ruff Linting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Maybe use v4? That's the latest version available and `tabpfn-server` is using v4 too.